src/components/primitives/Download/Download.tsx

```diff
@@ -1,42 +1,68 @@
 import * as React from "react";
-
-import styled, { withTheme } from "styled-components";
+import styled, { ThemeContext, withTheme } from "styled-components";
+import { getFontFromTheme } from "../../../theme/provider";
 
 export interface ComponentProps {
   href: string;
   fileName?: string | React.ReactNode;
+  darkTheme?: boolean;
 }
 
 const StyledDiv = styled.div<ComponentProps>`
-  font-size: 16px;
-  display: flex;
-  flex-direction: row;
-  flex-wrap: nowrap;
-  justify-content: space-between;
+  font-size: 1.1em;
+  diplay: inline-block;
+  width: 300px;
+  margin: 10px;
+  font-family: ${(props) => getFontFromTheme(props.theme).fontFamily};
+  .download {
+    background: ${(props) => {
+      return props.darkTheme ? props.theme.body.gray1 : props.theme.body.white;
+    }};
+    color: ${(props) => {
+      return props.darkTheme ? props.theme.body.white : props.theme.body.black;
+    }};
+    border-radius: ${(props) => props.theme?.buttonRadius || 0}px;
+    padding: 16px 10px 16px 20px;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    curson: pointer;
+    transition: all 0.3s;
+    &:hover {
+      background: ${(props) => {
+        const darkHover = props.theme.body.gray2;
+        const ligthHover = props.theme.body.gray3;
+        return props.darkTheme ? darkHover : ligthHover
+      }};
+    }
+    .icon {
+      margin-left: 30px;
+    }
+  }
 `;
 
 export const Download: React.FC<ComponentProps> = (props) => {
   const { href, fileName } = props;
+  const theme = React.useContext(ThemeContext);
 
   const name = React.useMemo(() => {
     return fileName || href.split("/").pop();
   }, [href, fileName]);
 
   return (
     <StyledDiv {...props}>
-      <a download href={href} target="_blank">
+      <a className="download" download href={href} target="_blank">
```

Comment:
imho its better to use more precise classes link `download-btn` or something like that